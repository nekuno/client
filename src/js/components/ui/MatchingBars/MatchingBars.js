import PropTypes from 'prop-types';
import React, { Component } from 'react';
import styles from './MatchingBars.scss';
import ProgressBar from "../ProgressBar/ProgressBar";
import translate from '../../../i18n/Translate';

@translate('MatchingBars')
export default class MatchingBars extends Component {

    static propTypes = {
        matching  : PropTypes.number,
        similarity: PropTypes.number,
        condensed : PropTypes.bool,
    };

    render() {
        const {matching, similarity, condensed, strings} = this.props;

        const className = condensed ? styles.matchingBarsCondensed : styles.matchingBars;
        return (
            <div className={className}>
                    <div className={styles.progressBarTitle}>{strings.compatible}&nbsp;</div>
                    <div className={styles.progressBar}>
                        <ProgressBar percentage={matching} size={'small'} strokeColor={'#756EE5'} background={'white'}/>
                </div>
                    <div className={styles.progressBarTitle}>{strings.similar}&nbsp;</div>
                    <div className={styles.progressBar}>
                        <ProgressBar percentage={similarity} size={'small'} strokeColor={'#756EE5'} background={'white'}/>
                    </div>
            </div>
        );
    }
}

MatchingBars.defaultProps = {
    strings   : {
        compatible: 'Compatibilidad',
        similar   : 'Similaridad',
        condensed : false,
    },
    matching  : 0,
    similarity: 0,
    condensed : false
};