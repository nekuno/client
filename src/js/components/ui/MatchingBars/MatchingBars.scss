@import '../../../../scss/_partials/colors';

.matching-bars {
  margin-top: 16px;
  padding: 0 5%;

  .progress-bar-title {
    display: inline-block;
    font-size: 8px;
    height: 33px;
    line-height: 33px;
    width: 50%;
    text-align: left;

    @media only screen and (min-width: 320px) {
      font-size: 10px;
    }
    @media only screen and (min-width: 370px) {
      font-size: 12px;
    }
  }

  .progress-bar {
    display: inline-block;
    width: 40%;
  }
}

.matching-bars-condensed {
  margin-top: 16px;
  padding: 0 5%;

  .progress-bar-title {
    display: inline-block;
    font-size: 8px;
    height: 33px;
    line-height: 33px;
    width: 100%;
    margin-bottom: -10px;

    @media only screen and (min-width: 320px) {
      font-size: 10px;
    }
    @media only screen and (min-width: 370px) {
      font-size: 12px;
    }
  }

  .progress-bar {
    display: inline-block;
    width: 100%;
  }
}