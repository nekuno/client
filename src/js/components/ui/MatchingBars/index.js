import PropTypes from "prop-types";
import React, { Component } from "react";
import styles from "./MatchingBars.scss";
import ProgressBar from "../ProgressBar";
import translate from "../../../i18n/Translate";

@translate("MatchingBars")
class MatchingBars extends Component {
	static propTypes = {
		matching: PropTypes.number,
		similarity: PropTypes.number,
		background: PropTypes.string,
		// Injected by @translate:
		strings: PropTypes.object
	};

	render() {
		const { matching, similarity, strings } = this.props;

		return (
			<div className={styles.matchingBars}>
				<div className={styles.progressBarTitle}>
					{strings.compatible}&nbsp;
				</div>
				<div className={styles.progress}>
					<ProgressBar percentage={matching} /*size={'small'} strokeColor={'#756EE5'} background={background}*/ />
					<div className={styles.matchingPercentage}>{Math.round(matching) + ' %'}</div>
				</div>
				<div className={styles.progressBarTitle}>
					{strings.similar}&nbsp;
				</div>
				<div className={styles.progress}>
					<ProgressBar percentage={similarity} /*size={'small'} strokeColor={'#756EE5'} background={background}*/ />
					<div className={styles.matchingPercentage}>{Math.round(similarity) + ' %'}</div>
				</div>
			</div>
		);
	}
}

MatchingBars.defaultProps = {
	strings: {
		compatible: "Compatibilidad",
		similar: "Similaridad",
	},
	matching: 0,
	similarity: 0,
	condensed: false,
	background: "white"
};

export default MatchingBars;
