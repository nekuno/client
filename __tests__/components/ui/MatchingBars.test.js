import React from 'react';
import MatchingBars from '../../../src/js/components/ui/MatchingBars/MatchingBars.js';
import styles from '../../../src/js/components/ui/MatchingBars/MatchingBars.scss';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Test MatchingBars component', () => {
    const mockFn = jest.fn();
    const matchingbars = shallow(<MatchingBars onClickHandler={mockFn}>Lorem ipsum</MatchingBars>);

    it('should be defined', () => {
        expect(matchingbars).toBeDefined();
    });
    it('should render correctly', () => {
        expect(matchingbars).toMatchSnapshot();
    });
    it('MatchingBars click event', () => {
        matchingbars.find(`.${styles.matchingbars}`).simulate('click');
        expect(mockFn).toHaveBeenCalledTimes(1);
    });
});
