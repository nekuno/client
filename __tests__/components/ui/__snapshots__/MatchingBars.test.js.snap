// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MatchingBars component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TranslationComponent
    background="white"
    condensed={false}
    matching={0}
    onClickHandler={[MockFunction]}
    similarity={0}
    strings={
      Object {
        "compatible": "Compatibilidad",
        "condensed": false,
        "similar": "Similaridad",
      }
    }
  >
    Lorem ipsum
  </TranslationComponent>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "background": "white",
      "children": "Lorem ipsum",
      "condensed": false,
      "locale": undefined,
      "matching": 0,
      "onClickHandler": [MockFunction],
      "similarity": 0,
      "strings": Object {
        "compatible": "Compatibilidad",
        "condensed": false,
        "similar": "Similaridad",
      },
    },
    "ref": null,
    "rendered": "Lorem ipsum",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "background": "white",
        "children": "Lorem ipsum",
        "condensed": false,
        "locale": undefined,
        "matching": 0,
        "onClickHandler": [MockFunction],
        "similarity": 0,
        "strings": Object {
          "compatible": "Compatibilidad",
          "condensed": false,
          "similar": "Similaridad",
        },
      },
      "ref": null,
      "rendered": "Lorem ipsum",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
